packages/core/src/publicCredential/PublicCredential.ts

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (c) 2018-2022, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import type { HexString } from '@polkadot/util/types'
+import type { AccountId } from '@polkadot/types/interfaces'
+import type { PublicCredentialsCredentialsCredential } from '@kiltprotocol/augment-api'
+import type {
+  DidUri,
+  IAssetClaim,
+  ICType,
+  IDelegationNode,
+  IPublicCredential,
+  IPublicCredentialInput,
+  PartialAssetClaim,
+} from '@kiltprotocol/types'
+
+import { blake2AsHex } from '@polkadot/util-crypto'
+import { ConfigService } from '@kiltprotocol/config'
+import * as AssetDid from '@kiltprotocol/asset-did'
+import * as Did from '@kiltprotocol/did'
+import { DataUtils, SDKErrors } from '@kiltprotocol/utils'
+
+import { verifyClaimAgainstSchema } from '../ctype/CType.js'
+import { toChain as publicCredentialToChain } from './PublicCredential.chain.js'
+
+/**
+ * Calculates the ID of a [[IPublicCredentialInput]], to be used to retrieve the full content from the blockchain.
+ *
+ * The ID is formed by first concatenating the SCALE-encoded [[IPublicCredentialInput]] with the SCALE-encoded [[DidUri]] and then Blake2b hashing the result.
+ *
+ * @param credential The input credential object.
+ * @param attester The DID of the credential attester.
+ * @returns The credential ID.
+ */
+export function computeId(
```

Comment:
I don’t see many `compute*` function in the SDK, would the consistent name be `getIdFromCredential()`?

Reply:
But `get` sounds like it's in there and we retrieve it. What we do is we re-calculate it based on the other properties of the credentials. Compute might not be the best, but I don't think get is the right word either here.

Reply:
We already have `getHashForSchema()` which does the same. I did misremembered `for` and typed `from`, so my updated suggestions is `getIdForCredential()`

Reply:
Good point. I changed it then, but I still don't like the naming 😄

Reply:
https://github.com/KILTprotocol/sdk-js/pull/677/commits/023ad1c369410f4be90a93e0c4363bf2b8759885#diff-0ed53062bda598489034fe7ca13c690382a38738d8e6ea23ebfc8b3a6a3b5880R39

Reply:
I also had a long discussion with Timo on a similar topic :) Consistency is important though.

Reply:
I will resolve this conversation now, as we seem to have reached mutual understanding and an agreement.